### PR TITLE
[Core] Propagate ownership information into object directory and object manager

### PR DESCRIPTION
**Pull request by zhuohan123**
## Why are these changes needed?

Propagate ownership information into the object directory and the object manager. This step is necessary building an owner-based object directory.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

File: src/ray/object_manager/object_manager.cc [hunk 1]
```diff
@@ -469,14 +475,15 @@ void ObjectManager::Push(const ObjectID &object_id, const ClientID &client_id) {
...
-    std::shared_ptr<rpc::ObjectManagerClient> rpc_client) {
+    const UniqueID &push_id, const ObjectID &object_id, const rpc::Address &owner_address,
+    const ClientID &client_id, uint64_t data_size, uint64_t metadata_size,
+    uint64_t chunk_index, std::shared_ptr<rpc::ObjectManagerClient> rpc_client) {
   double start_time = absl::GetCurrentTimeNanos() / 1e9;
   rpc::PushRequest push_request;
   // Set request header
   push_request.set_push_id(push_id.Binary());
   push_request.set_object_id(object_id.Binary());
+  *push_request.mutable_owner_address() = owner_address;
```
Comment (stephanie-wang) [inline src/ray/object_manager/object_manager.cc:486]: I think you can also do `mutable_owner_address()->CopyFrom(...)`.

File: src/ray/object_manager/object_manager.h [hunk 2]
```diff
@@ -138,6 +138,7 @@ class ObjectManager : public ObjectManagerInterface,
   /// \param chunk_index Chunk index of this object chunk, start with 0
   /// \param rpc_client Rpc client used to send message to remote object manager
   ray::Status SendObjectChunk(const UniqueID &push_id, const ObjectID &object_id,
+                              const rpc::Address &owner_address,
```
Comment (stephanie-wang) [inline src/ray/object_manager/object_manager.h:141]: Can you document the arg?

**Review by stephanie-wang — APPROVED** [review]

**Comment by ericl** [comment]
Is this passing tests?

**Comment by ericl** [comment]
LINT is still failing.

**Comment by zhuohan123** [comment]
> LINT is still failing.

Fixed!

**Comment by ericl** [comment]
Please fix the merge conflict.

**Comment by ericl** [comment]
Looks like there are some more conflicts

**Comment by ericl** [comment]
@zhuohan123

**Comment by zhuohan123** [comment]
Close this PR since all the changes are included in #9735 now.

**Pull request closed without merging**